### PR TITLE
Credential recovery

### PR DESCRIPTION
**Pull request by AngeloCG97**
## Credential recovery

#58 

### What does this PR do?

- Code desing to email
- Generate and encripted tempSecret
- Update tempSecret
- Create actions and querrys to change password
- Validate current password and change to new password
- Coding design recovery password page
- Send and coding design email to change password when is ok
- Add validations to fields

### How should this be tested?

- Run make run
- Go to localhost:3000
- Go to recovery credentials

### CODE REVIEW

File: hapi/src/routes/change-password/change-password.handler.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    })
+
+    if (currentPasswordIsOk) {
+      const encripnewPassword = await new Promise((resolve) => {
+        bcrypt.hash(input.newPassword, saltRounds, function (err, hash) {
+          if (!err) resolve(hash)
+        })
+      })
+
+      const user = await userApi.setSecret({ email: {_eq: input.email} }, encripnewPassword)
```
Comment (JustinCast) [inline hapi/src/routes/change-password/change-password.handler.js:28]: Take a look at this prettier suggestion

File: hapi/src/routes/change-password/change-password.handler.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
```
Comment (JustinCast) [inline hapi/src/routes/change-password/change-password.handler.js:29]: Prettier suggestion here

File: hapi/src/routes/change-password/change-password.handler.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      if (user) await mailUtils.sendConfirmMessage(
+        input.email,
+        'Lifebank Change Password',
```
Comment (JustinCast) [inline hapi/src/routes/change-password/change-password.handler.js:32]: Prettier suggestion here

File: hapi/src/routes/change-password/change-password.handler.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        'Change Password',
```
Comment (JustinCast) [inline hapi/src/routes/change-password/change-password.handler.js:33]: Prettier suggestion here

File: hapi/src/routes/change-password/change-password.handler.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        'Your password has been successfully changed, thank you for donating life'
```
Comment (JustinCast) [inline hapi/src/routes/change-password/change-password.handler.js:34]: Prettier suggestion here

File: hapi/src/routes/change-password/change-password.handler.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+      )
```
Comment (JustinCast) [inline hapi/src/routes/change-password/change-password.handler.js:35]: Prettier suggestion here

File: hapi/src/routes/change-password/change-password.handler.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
```
Comment (JustinCast) [inline hapi/src/routes/change-password/change-password.handler.js:36]: Prettier suggestion here

File: hapi/src/routes/change-password/change-password.handler.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      return {
+        success: true
+      }
+    } else {
+
```
Comment (JustinCast) [inline hapi/src/routes/change-password/change-password.handler.js:41]: Prettier suggestion here

File: hapi/src/routes/credentials-recovery/credentials-recovery.handler.js [hunk 9]
```diff
@@ -1,17 +1,33 @@
...
+
+    const encripTempSecret = await new Promise((resolve) => {
+      bcrypt.hash(tempSecret, saltRounds, function (err, hash) {
+        if (!err) resolve(hash)
+      })
     })
 
+    const user = await userApi.setSecret(
+      { email: { _eq: input.email } }, 
+      encripTempSecret
```
Comment (JustinCast) [inline hapi/src/routes/credentials-recovery/credentials-recovery.handler.js:22]: Prettier suggestion here

File: webapp/src/components/CredentialsRecovery/CredentialsRecovery.js [hunk 10]
```diff
@@ -96,28 +124,65 @@ const CredentialsRecovery = ({ overrideBoxClass, overrideLabelClass }) => {
...
         ...user
       }
     })
+    setValidEmailFormat(false)
   }
 
   useEffect(() => {
     if (error) {
-      setErrorMessage(error.message.replace('GraphQL error: ', ''))
+      if(error.message === `GraphQL error: Cannot read property 'account' of undefined`)
```
Comment (JustinCast) [inline webapp/src/components/CredentialsRecovery/CredentialsRecovery.js:156]: What if the version changes and due to that, the error is now different?

**Review by JustinCast — CHANGES_REQUESTED** [review]

**Review by JustinCast — APPROVED** [review]

**Review by JeanVegaD — APPROVED** [review]

**Pull request merged**